interpreter/cling/tools/plugins/clad/CMakeLists.txt

```diff
@@ -49,11 +49,7 @@ if (LLVM_FORCE_USE_OLD_TOOLCHAIN)
   list(APPEND _clad_extra_cmake_args -DLLVM_FORCE_USE_OLD_TOOLCHAIN=${LLVM_FORCE_USE_OLD_TOOLCHAIN})
 endif(LLVM_FORCE_USE_OLD_TOOLCHAIN)
 
-if (CLING_INCLUDE_TESTS)
-  list(APPEND _clad_extra_cmake_args -DLLVM_EXTERNAL_LIT=${LLVM_MAIN_SRC_DIR}/utils/lit/lit.py)
-else()
-  list(APPEND _clad_extra_cmake_args -DCLAD_BUILD_STATIC_ONLY=ON)
-endif()
+list(APPEND _clad_extra_cmake_args -DCLAD_BUILD_STATIC_ONLY=ON)
```

Comment:
How does this disable the `Clad` test?  In the commit it mentions "now results in a cycle after
commit https://github.com/root-project/root/commit/fe6c671e856d8fd2a1a3b39d116499a0d8a86aee made clang depend on clad."   Which does "depend on clad" means "depend on the clad test"?

> Since we never run the Clad tests, never build them.

@vgvassilev Are they run in a different context?

Reply:
> How does this disable the Clad test?

It disables building the Clad tests (that are never run), see https://github.com/vgvassilev/clad/blob/6cc83eeca4a42c862493898a892b7c57abf55109/CMakeLists.txt#L317

> In the commit it mentions "now results in a cycle after commit https://github.com/root-project/root/commit/fe6c671e856d8fd2a1a3b39d116499a0d8a86aee made clang depend on clad." Which does "depend on clad" means "depend on the clad test"?

The linked commit adds `add_dependencies(clang clad)`, where `clad` is the target created by `ExternalProject_Add`. I don't know how to better describe this, to be honest, if the added code is that clear and literally only one click away. The reason that Clad tests (built in `ExternalProject_Add`, so contained in target `clad`) depend on Clang is because of this line: https://github.com/vgvassilev/clad/blob/6cc83eeca4a42c862493898a892b7c57abf55109/CMakeLists.txt#L334

> @vgvassilev Are they run in a different context?

The Clad unit tests are run as part of https://github.com/vgvassilev/clad/. To be fully clear, this change *does not* touch any ROOT tests that exercises Clad from our side.

Reply:
Yes, this change seems reasonable to me.

Reply:
> The reason that Clad tests (built in ExternalProject_Add, so contained in target clad) depend on Clang is because of this line: [vgvassilev/clad@6cc83ee/CMakeLists.txt#L334](https://github.com/vgvassilev/clad/blob/6cc83eeca4a42c862493898a892b7c57abf55109/CMakeLists.txt#L334)

Humm ... that seems to be a real problem ... for clad.

Semantically there is no cycle:
   * clang depends on clad (clang needs to be linked against `cladPlugin cladDifferentiator`
   * clad **tests** depends on clad and clang ("Our unittests need to use a compiler supported by clad")
   
However the technique used to enforce the second bullet: [vgvassilev/clad@6cc83ee/CMakeLists.txt#L334](https://github.com/vgvassilev/clad/blob/6cc83eeca4a42c862493898a892b7c57abf55109/CMakeLists.txt#L334) is too broad.

Reply:
Yes, you are right, there should be no cycle. But there is when building as a Cling plugin because (as I already wrote above...) we use `ExternalProject_Add` which mixes all building (library and tests) into the single `clad` target. Let me stress again that the Clad tests are never executed in that situation anyway, so I don't see a reason to run into problems building them.

> However the technique used to enforce the second bullet: [vgvassilev/clad@6cc83ee/CMakeLists.txt#L334](https://github.com/vgvassilev/clad/blob/6cc83eeca4a42c862493898a892b7c57abf55109/CMakeLists.txt#L334) is too broad.

No, this seems to work fine because `CMAKE_CXX_COMPILER` is set only before the `unittests` and `tests` directory. This is also evidenced by the fact that it's only a problem with `CLING_INCLUDE_TESTS`, and that it only complains about building the tests.

Given that we are starting to discuss the Clad build system here and I don't see anything to do on the ROOT / Cling side, I'm going to resolve this thread.

Reply:
> we use ExternalProject_Add which mixes all building (library and tests) into the single clad 

Thanks for that reminder.